test/templates/unit_test.tpl

```diff
@@ -14,7 +14,7 @@ env:
 jobs:
   test:
     name: {{.Name}}
-    runs-on: ubuntu-22.04
+    runs-on: gh-hosted-runners-4cores-1
```

Comment:
We run unit tests on 4cores and not 16 because we usually run them sequentially and they don't normally create too many threads?

Reply:
@rohit-nayak-ps but that's a good point, I have move `.github/workflows/unit_race.yml` to 16 cores.